### PR TITLE
Bundle Netty Maven Jars into an independant internal feature

### PR DESCRIPTION
**Pull request by hutchig**
Puts the maven assets of Netty that grpc uses into a bundle.
(Does not yet make grpc use them from here though)

### CODE REVIEW

**Comment by hutchig** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature:3]: I don't think you want this line on a new feature, it's only for old GA features

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature:7]: edition=full until the feature is moved to beta

File: dev/io.openliberty.netty.internal/bnd.bnd
Comment (tevans78) [inline dev/io.openliberty.netty.internal/bnd.bnd:33]: What does `-split-package:=merge-first` do? Split packages are to be avoided if possible.

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature
Comment (hutchig) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature:3]: @bjhargrave do you know the best practice for this now? feature props has a lot of stuff that I am no SME for and I am wary of not doing what most other example do.

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature
Comment (hutchig) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature:7]: I would prefer to leave it at the correct value. Why do you care?

File: dev/io.openliberty.netty.internal/bnd.bnd
Comment (hutchig) [reply dev/io.openliberty.netty.internal/bnd.bnd:33]: This is needed until the netty packages are removed from grpc.
This is just a draft PR at the moment.

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature
Comment (bjhargrave) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature:3]: Sorry, I don't know anything about what `WLP-DisableAllFeatures-OnConflict` means in OL. It is not a Bnd thing. @tjwatson may be able to provide advice here.

Comment (tjwatson) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature:3]: For new features do not set `WLP-DisableAllFeatures-OnConflict`.  This is for old features to preserve the behavior for when conflicts are found.  New features should not set this header so our build will automatically set the header value to the default (which is to force conflicts to disable all features and produce a message that the server config is not supported).  For details on the previous design issue see #13928

File: dev/io.openliberty.netty.internal/bnd.bnd
Comment (bjhargrave) [reply dev/io.openliberty.netty.internal/bnd.bnd:33]: You need a semicolon before `-split-packages`

```properties
Export-Package: \
  io.netty.*;version=${nettyVersion};-split-package:=merge-first
```

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature
Comment (hutchig) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature]: Deleted

File: dev/io.openliberty.netty.internal/bnd.bnd
Comment (hutchig) [reply dev/io.openliberty.netty.internal/bnd.bnd]: Added

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature
Comment (hutchig) [reply dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.netty.internal.feature:6]: As discussed, I believe that is just a example, not guidance,  in the doc. Without some reason, I am happy to test leaving it correct from the start.

**Comment by hutchig** [comment]
Given that the 'implementation' project is going to be called io.openliberty.netty
and, as yet, we are not exposing the netty API to apps and no 'spec' (so there is no API-impl split)
It makes sense to have this as `io.openliberty.io.netty` @tevans78 
as although this is not a 1:1 bundler project - that is it's main job and having
the io.netty in the name makes it easily recognizable as that
(whereas io.openliberty.netty.internal looks as if could be what io.openliberty.netty is doing, which it is not)
I took the original .internal name from the grpc structure where it made more sense.

**Comment by hutchig** [comment]
@wtlucy I have split this into handler.ssl.* and non ssl now like grpc internal client and grpc client thirdparty but wondered if we should leave it as a draft/non-merged until we get UFO approval for changing the codebase?
This draft PR does not change any of the current files is easy to source though to build on top of for now. We can chat  in the scrum.

**Comment by hutchig** [comment]
#build

**Comment by wtlucy** [comment]
@hutchig these projects and internal features/bundles look good to me - with the current split it will be easy for the grpc features to consume them. Like I said on the scrum, I think it's fine for us to merge this type of code prior to going through the design/UFO approval process, since the features you're adding are `noship` and won't add or change any released functionality.

**Comment by hutchig** [comment]
#build

**Comment by hutchig** [comment]
Hi @wtlucy I think I am happy enough with this now to merge it if you approve on review and I have checked the grpc fat tests pass ok.

**Comment by tevans78** [comment]
@hutchig please squash the commits down before merging

File: dev/io.openliberty.io.netty.ssl/bnd.bnd
Comment (tevans78) [inline dev/io.openliberty.io.netty.ssl/bnd.bnd:2]: wrong date

File: dev/io.openliberty.io.netty/bnd.bnd
Comment (wtlucy) [inline dev/io.openliberty.io.netty/bnd.bnd:24]: This package shouldn't need to be excluded (monitor isn't on the buildpath)

**Review by wtlucy — CHANGES_REQUESTED** [review]

File: dev/io.openliberty.io.netty/bnd.bnd
Comment (hutchig) [reply dev/io.openliberty.io.netty/bnd.bnd:24]: Agreed, I will check both the bnd files.

Comment (hutchig) [reply dev/io.openliberty.io.netty/bnd.bnd]: Done

**Comment by hutchig** [comment]
#build

**Review by tevans78 — APPROVED** [review]
LGTM

**Comment by hutchig** [comment]
#run-libby-bot

**Comment by hutchig** [comment]
Do not merge until it lines up with build from source in https://github.com/OpenLiberty/open-liberty/pull/15887

**Comment by hutchig** [comment]
How were these tested?
They do not have currently active code, they were tested using the automated CI and fire regression builds/checks and FAT buckets. Subsequent work will make use of these files.

**Comment by wtlucy** [comment]
fixes #15668

**Pull request merged**
